Automattic/wp-calypso
# Staging Sites: fix card buttons and long URL on mobile display

* Closes https://github.com/Automattic/dotcom-forge/issues/2210

## Proposed Changes

This PR addresses inconsistencies in the "Manage staging site" and "Delete staging site" on mobile devices. Some issues it addresses are:

* if a staging site has a very long URL, it gets cut off on mobile devices on a staging card https://wordpress.com/hosting-config/{productionsite}
* on some mobile devices and in some browsers, the buttons "Manage staging site" and "Delete staging site" are not getting wrapped and overflow 

## Testing Instructions

**To test long URL being cut off**

This is mostly an edge case as it happens only when the URL is very long:

* Create a production Atomic site with a very long domain (40 characters is a good size, for example, `kathasasitewithaverylongdomainfortesting.wpcomstaging.com`)
* Go to `https://wordpress.com/hosting-config/{yourproductionsite}`
* Create a staging site that should also have a long URL
* Check the view of the long domain on mobile (using Chrome Dev Tools)
* Confirm that the URL does not overflow and goes on the next line:

<img width="1004" alt="Screenshot 2023-05-17 at 1 23 22 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/7b2497b9-ac99-417f-a54c-3c545f0d2b5e">

This is what it looked like **before**:

<img width="726" alt="Screenshot 2023-05-17 at 1 21 56 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/9b57f175-50c3-485c-978c-f117d191e41d">

**Issue 2 - Updated**:

For now, we decided to keep the original design on the desk

* Create a production Atomic site with a very long domain (40 characters is a good size, for example, `kathasasitewithaverylongdomainfortesting.wpcomstaging.com`)
* Go to `https://wordpress.com/hosting-config/{yourproductionsite}`
* Create a staging site
* Check the buttons "Manage staging site" and "Delete staging site"  maintain their original design on the desktop
* Switch to mobile view (using Chrome Dev Tools) is fine
* Check that the buttons "Manage staging site" and "Delete staging site" are stacked on the top of each other and fill the screen like so:

<img width="955" alt="Screenshot 2023-05-17 at 4 37 37 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/b5f84972-5d51-48d7-b92d-d770335ae158">

**Previous notes on possible design solution**:

Another part of the issue,  it was reported that on some screens the buttons "Manage staging site" and "Delete staging site" either overflow or the distance between the buttons and the sides of the screen is consistent on different mobile devices and browsers. You can see an example of the issue happening on this screencast:

https://github.com/Automattic/wp-calypso/assets/25575134/a15a619a-383c-4a0b-88fa-bb7396745b9b

This is something that is expected at the moment because we are using `display: 'flex'` for the container that has the buttons so if they fit on the screen, they will be placed there like so: `expected margin on the left + button + gap between buttons + second button + whatever space is left`. The key part of this is `whatever space is left` as this depends on the screen size of the used device: on some devices, there is more space and on some less which creates inconsistency of how much space is on each side of the button.

One option that we can do is to stack the buttons one on top of the other either by using `flex-wrap: wrap` on the parent container and `flex-grow` on the children. This change will affect how the buttons are displayed on the **bigger screens** though:

<img width="1323" alt="Screenshot 2023-05-17 at 3 18 56 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/536b66ea-bea2-4808-aed9-2a1045d0c83c">

<img width="995" alt="Screenshot 2023-05-17 at 3 18 28 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/d6553360-884f-40c2-87fb-31c98c579347">

<img width="1206" alt="Screenshot 2023-05-17 at 3 18 50 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/2983d78b-efb6-4060-8b3a-5f5880327115">

Another option would be to use a media query to also stack the buttons on top of each other but keep the size that they have now. This would preserve the original design but would be a less responsive solution:

<img width="1080" alt="Screenshot 2023-05-17 at 3 24 27 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/1b866265-d4b6-40c3-bd61-b1e40f5f3c72">

The bigger screen remains unaffected in this case and is what it is now. 

@javierarce @danielbachhuber - what do you think would be a better approach here? 


## Pre-merge Checklist
